drivers/pinctrl/Kconfig.imx

```
@@ -29,3 +29,9 @@ config MCUX_XBARA
 	depends on HAS_MCUX_XBARA
 	help
 	  Enable the MCUX XBARA driver.
+
+config MCUX_XBARB
+	bool "MCUX XBARB driver"
+	depends on HAS_MCUX_XBARB
```

Remove the config HAS_MCUX_XBARB from Kconfig.mcux and
```suggestion
	depends on DT_HAS_NXP_MCUX_XBAR_ENABLED
```
@decsny This what you mean by depending on DT? Should it also be done for XBARA above?

---

yes these HAS_MCUX_XXX kconfigs are super old legacy from before DT was a thing